### DISCUSSION TITLE
Get the primaryKey on custom displays

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#19877] [Feature Requests]
### Summary

Currently, the primaryKey is provided on interfaces.
But not on displays.

### Motivation

It happens that the display has to be formatted according to the value of another field of the row.
Hence we need to api.get that field based on the primaryKey (as the props.value is itself not an unique identifier).

### Detailed Design

Send the primaryKey to the custom extension

### Requirements List

Send the primaryKey to the custom extension

### Drawbacks

None

### Alternatives

None

### Adoption Strategy

Immediate

**Comment by rijkvanzanten** [comment]
> It happens that the display has to be formatted according to the value of another field of the row.
Hence we need to api.get that field based on the primaryKey (as the props.value is itself not an unique identifier).

This feels like an easy way to DDoS yourself! You'd end up making that request for each item visible in a layout, which could be tens of thousands!

**Reply by Pachat** [reply]
Since I have control over the number of items I display (<20 at this location in the application), this is not a problem.

But as you said, if you really want to avoid auto-DDoS on a large number of lines,
expanding the feature request to any other column in the collection would fit the need perfectly.
It would be much more direct and efficient.

**Comment by max-mayorov** [comment]
I'm also interested in this, trying to build a display extension in which render of a field will depend on some other field...

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
